packages/next/src/client/components/app-router.tsx

```diff
@@ -184,41 +181,23 @@ function Router({
       back: () => window.history.back(),
       forward: () => window.history.forward(),
       prefetch: async (href) => {
-        const hrefWithBasePath = addBasePath(href)
-
         // If prefetch has already been triggered, don't trigger it again.
-        if (
-          prefetched.has(hrefWithBasePath) ||
-          (typeof window !== 'undefined' && isBot(window.navigator.userAgent))
-        ) {
+        if (isBot(window.navigator.userAgent)) {
           return
         }
-        prefetched.add(hrefWithBasePath)
-        const url = new URL(hrefWithBasePath, location.origin)
+        const url = new URL(addBasePath(href), location.origin)
         // External urls can't be prefetched in the same way.
         if (isExternalURL(url)) {
           return
         }
-        try {
-          const routerTree = window.history.state?.tree || initialTree
-          const serverResponse = await fetchServerResponse(
+
+        // @ts-ignore startTransition exists
+        React.startTransition(() => {
```

Comment:
nit: can you extend it at the global level in a def file?

Reply:
This one is covered in the latest `@types/react`, still has to be upgraded, not super important right now 👍